### PR TITLE
OM-3131: Add Career Page Settings Form, Preview

### PR DESCRIPTION
**Pull request by amar-hunar**
This PR addresses part of [OM-3131](https://hunar.atlassian.net/browse/OM-3131)

## Features:

- Show dialog to add career page settings after successful company creation
- Add form to populate career page settings
- Add career page preview

## Technical:

- The following api hooks are added:
  - `useGetCompanies`
  - `useUploadCareerPageAsset`
  - `useAddCareerPageSettings`
- The following components are added:
  - `CareerPageSetupDialog`
  - `CompanyCareerPagePreview`
  - `CompanyCareerJQCardPreview`
  - `UploadButton` (The old MUI version is removed. A new component is built using Chakra UI)
  - `UploadInputPreview`

### CODE REVIEW

File: src/components/company/CareerPageSetupDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+export const CareerPageSetupDialog = ({
+    isOpen,
+    onCloseClick,
+    onProceedClick
+}: CareerPageSetupDialogProps) => {
+    return (
+        <Modal isOpen={isOpen} onClose={onCloseClick}>
+            <ModalOverlay />
```
Comment (misss-popcorn) [inline src/components/company/CareerPageSetupDialog.tsx:25]: can we make a common modal? Refer dashboard

File: src/components/company/CompanyCareerJQCardPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                    fontSize="sm"
+                    sx={{
+                        overflow: 'hidden',
+                        textOverflow: 'ellipsis',
+                        display: '-webkit-box',
+                        WebkitLineClamp: '2',
+                        WebkitBoxOrient: 'vertical'
+                    }}
+                >
+                    Age - 18 - 28 Years Education Qualification - Minimum 12th
```
Comment (misss-popcorn) [inline src/components/company/CompanyCareerJQCardPreview.tsx:33]: Put it in literals

File: src/components/company/CompanyCareerPageForm.tsx [hunk 3]
```diff
@@ -1,27 +1,515 @@
...
+
+    const onFileRemove = (fieldName: string) => {
+        updateForm({ [fieldName]: '' });
+        updateFieldErrorState({
+            fieldName: fieldName as keyof CareerPageFormProps,
+            fieldValue: ''
+        });
+    };
+
+    const addSettings = () => {
```
Comment (misss-popcorn) [inline src/components/company/CompanyCareerPageForm.tsx:211]:
```suggestion
    const submitSettings = () => {
```

Comment (misss-popcorn) [reply src/components/company/CompanyCareerPageForm.tsx:211]: use prefix, `submit` or `save` for api call methods

File: src/components/company/CompanyCareerPageForm.tsx [hunk 4]
```diff
@@ -1,27 +1,515 @@
...
+                            name="primaryColor"
+                            value={form.primaryColor}
+                            onChange={onFormFieldChange}
+                        />
+                        <HelperText
+                            hasError={formErrorState.primaryColor}
+                            errorMsg={ErrorMsg.hexColor()}
+                        />
+                    </FormControl>
+                    <FormControl
```
Comment (misss-popcorn) [inline src/components/company/CompanyCareerPageForm.tsx:343]: refactor it

File: src/hooks/apiHooks/careerPage/useAddCareerPageSettings.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        ApiError,
+        AddCareerPageSettingsProps
+    >(({ params, requestBody }: AddCareerPageSettingsProps) => {
+        return settings
+            .post({
+                params,
+                body: requestBody
+            })
+            .then((response: AddCareerPageSettingsResponse) => {
+                return response;
```
Comment (misss-popcorn) [inline src/hooks/apiHooks/careerPage/useAddCareerPageSettings.ts:32]: can we return promise

Comment (amar-hunar) [reply src/hooks/apiHooks/careerPage/useAddCareerPageSettings.ts:32]: Need to discuss this solution on call.

Comment (amar-hunar) [reply src/hooks/apiHooks/careerPage/useAddCareerPageSettings.ts:32]: To be done in a separate PR

**Pull request merged**